lib/wifi/portable/espressif/esp32_devkitc_esp_wrover_kit/aws_wifi.c

```diff
@@ -591,7 +591,23 @@ WIFIReturnCode_t WIFI_GetMAC( uint8_t * pucMac )
     /* Try to acquire the semaphore. */
     if( xSemaphoreTake( xWiFiSem, xSemaphoreWaitTicks ) == pdTRUE )
     {
-        esp_err_t ret = esp_wifi_get_mac(WIFI_MODE_STA, pucMac);
+        wifi_mode_t mode;
+        esp_err_t ret = esp_wifi_get_mode(&mode);
+        if( ret == ESP_OK )
+        {
+            if( mode == WIFI_MODE_STA )
```

Comment:
will a switch case be better in this?

Reply:
For many conditions, switch is better, but for just three condition, it won't make a difference.